contracts/fund/HighriseLandFund.sol

```diff
@@ -53,23 +55,25 @@ contract HighriseLandFund {
             (uint256, uint256, uint256)
         );
         require(expiry > block.timestamp, "Reservation expired");
-        require(msg.value >= cost);
+        require(msg.value == cost, "Amount sent does not match land price");
```

Comment:
We need to test this, I'm worried about mismatches here because of precision bugs between all the services.